logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamBatchWriterImpl.java

```diff
@@ -197,11 +196,12 @@ public long tryWrite() {
       copyExistingEventToBuffer();
     }
 
-    long result = claimBatchForEvents();
-    if (result >= 0) {
+    long position = claimBatchForEvents();
+    if (position >= 0) {
       try {
         // return position of last event
-        result = writeEventsToBuffer(claimedBatch.getBuffer());
+        writeEventsToBuffer(claimedBatch.getBuffer(), position);
+        position += eventCount - 1;
```

Comment:
didn't you wrote an test in the dispatcher where the claim returned the the position based on the fragment count?

Reply:
So I think claim will return already the last position? ANd you then add again the event count?

Reply:
Not quite. The Dispatcher tests verify that the **next** position increases based on the current fragment count. Claim returns the position of the first record. I think it makes more sense than returning the updated publisher position and then calculating the initial in the writers. The single writer had to subtract a frame length from the claimedPosition to get the actual position and the batch writer was replacing the position anyway using the position returned from `ClaimedFragmentBatch#nextFragment`.

Reply:
I have the feeling the api can still be improved. We dont need to do that now but just an Idea.

We could return a list of claimed fragments. The ClaimedFragment holds and buffer and a position, which the write can use. You can then iterate over the list write in each fragment the corresponding data and the related position and dont need to recalculate it by yourself. wdyt?

Reply:
I'm not sure I understand the goal. We used to do the same things in the writers, except in a more complicated way. Previously, the position returned by the Dispatcher was not useful for the batch writer (it was not being used except as a success/fail flag) and was overwritten by the last calculated position. It was also not directly used by the single writer since we had to subtract a frame length to get the actual position of the fragment. Now the single writer uses the position returned by the Dispatcher as is and, since we want to return last event's position in the batch writer, we do that by taking into account the number of events in the batch. 
I might be misunderstanding the idea but returning a list of claimed fragment is different than a batch fragment, if we wrote the first fragment and failed in the second, the first fragment wouldn't be aborted. That gives us different semantics in writing. Maybe we could write some additional logic to get around this somehow but I'm not sure I understand why we would. To avoid doing `position += eventCount - 1`? I'm not sure I see the benefit, but I might not be seeing the full picture.

Reply:
I think you misunderstood me. What I would like to have as an API is to not care where the position came from and how it is calculated.

What we currently doing is to claim a batch, which is just a buffer and go over our to write events and write them in the buffer we got. We calculating the position then based on index and the returned position.


What I like to have is the following:


```
for (ClaimedFragment fragment : claimedBatch.getFragments) {

 writeDateInto(fragment.buffer);
 setPositionForEvent(fragment.position)

}

```